stdlib/public/runtime/Casting.cpp

```diff
@@ -1115,6 +1115,14 @@ swift_dynamicCastMetatypeImpl(const Metadata *sourceType,
     }
     break;
 
+  case MetadataKind::Existential: {
+    auto targetTypeAsExistential = static_cast<const ExistentialTypeMetadata *>(targetType);
+    for (auto protocol : targetTypeAsExistential->getProtocols())
+      if (!swift_conformsToProtocol(sourceType, protocol.getSwiftProtocol()))
```

Comment:
There are a couple more things to check here:
* Whether the existential is class-bounded (`isClassBounded`)
* The superclass of the existential (`getSuperclassConstraint`)
* `@objc` protocols (`getSwiftProtocol()`). You might be able to use `swift::_conformsToProtocol` rather than `swift_conformsToProtocol`, because it handles the `ProtocolDescriptorRef` directly).

Reply:
Thank you, Doug! It turns out that `_conformsToProtocols` handles all of this anyway, and lets me ditch the loop. I added tests for all of these to make sure. Please let me know if you see anything else I should account for.